util/afero_test.go

```diff
@@ -20,17 +20,104 @@
 package util
 
 import (
+	"bytes"
 	"io/ioutil"
 	"os"
 	"testing"
 
 	"github.com/spf13/afero"
 )
 
-func TestCopyDiskToFs1(t *testing.T) {
-	if true {
-		return // XXX: remove me once this test passes
+var dirInputs = []struct {
+	srcDirFull  string
+	srcCopyRoot string
+	dstCopyRoot string
+	dstExpected string
+	force       bool
+}{
+	{"/tmp/foo/bar/baz/", "/tmp/foo/", "/", "/foo/bar/baz", false},
+	{"/tmp/zoo/zar/zaz/", "/tmp/zoo/zar", "/start/dir", "/start/dir/zar/zaz", false},
+	{"/foo", "/foo", "/", "/foo", false},
+	{"/foo", "/foo", "/", "/foo", true},
+}
+
+func TestCopyFs1(t *testing.T) {
+	for _, tt := range dirInputs {
+		src := afero.NewMemMapFs()
+		dst := afero.NewMemMapFs()
+
+		t.Run(tt.srcDirFull, func(t *testing.T) {
+			err := src.MkdirAll(tt.srcDirFull, 0700)
+			if err != nil {
+				t.Errorf("could not MkdirAll %+v", err)
+				return
+			}
+			err = CopyFs(src, dst, tt.srcCopyRoot, tt.dstCopyRoot, tt.force)
+			if err != nil {
+				t.Errorf("error copying source %s to dest %s", tt.srcCopyRoot, tt.dstCopyRoot)
+				return
+			}
+
+			isDir, err := afero.IsDir(dst, tt.dstExpected)
+			if err != nil {
+				t.Errorf("could not check IsDir: %+v", err)
+				return
+			}
+			if !isDir {
+				t.Errorf("expected directory tree %s to exist in dest", tt.dstExpected)
+				return
+			}
+		})
 	}
+}
+
+func TestCopyFs2(t *testing.T) {
+	tree := "/foo/bar/baz/"
+	var files = []struct {
+		path    string
+		content []byte
+	}{
+		{"/foo/foo.txt", []byte("foo")},
+		{"/foo/bar/bar.txt", []byte("bar")},
+		{"/foo/bar/baz/baz.txt", []byte("baz")},
+	}
+
+	src := afero.NewMemMapFs()
+	dst := afero.NewMemMapFs()
+
+	err := src.MkdirAll(tree, 0700)
+	if err != nil {
+		t.Errorf("could not MkdirAll: %+v", err)
+		return
+	}
+
+	for _, f := range files {
+		err = afero.WriteFile(src, f.path, f.content, 0600)
+		if err != nil {
+			t.Errorf("could not WriteFile: %+v", err)
+			return
+		}
+	}
+
+	err = CopyFs(src, dst, "", "", false)
```

Comment:
you can use the golang form:

if err := CopyFS(...); err != nil {

and you need a return in there to end the test early.